File: Swift/056ValidAnagram/Sources/ValidAnagram/ValidAnagram.swift
import Foundation

class ValidAnagram {
	func isAnagram(_ str1: String, _ str2: String) -> Bool {
		guard str2.count == str1.count else { return false }
		var str1Map = [Character: Int](), str2Map = [Character: Int]()
		for char in str1 {
			str1Map[char, default: 0] += 1
		}
		for char in str2 {
			str2Map[char, default: 0] += 1
		}
		for char in str1Map.keys where str1Map[char] != str2Map[char] {
			return false
		}
		return true
	}
}

File: C++/056ValidAnagram/ValidAnagram/src/Validanagram.cpp
#include "Validanagram.h"
#include <bits/stdc++.h>

void Validanagram::hello() {
  std::cout << "hello" << std::endl;
}

bool Validanagram::isAnagram(string str1, string str2) {
  if (str1.size() != str2.size())
    return false;
  map<char, int> str1Map;
  map<char, int> str2Map;
  for (char & i : str1) {
    if (str1Map.find(i) != str1Map.end()) {
      str1Map[i] = 1;
    } else {
      str1Map[i] = 1 + str1Map[i];
    }
  }
  for (char & i : str2) {
    if (str2Map.find(i) != str2Map.end()) {
      str2Map[i] = 1;
    } else {
      str2Map[i] = 1 + str2Map[i];
    }
  }

  for (const auto &item: str1Map) {
    if (item.second != str2Map.count(item.second))
      return false;
  }
  return true;
}
